framework-support/jobrunr-spring-boot-2-starter/src/main/java/org/jobrunr/scheduling/JobInterceptor.java

```diff
@@ -0,0 +1,53 @@
+package org.jobrunr.scheduling;
+
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.jobrunr.jobs.JobDetails;
+import org.jobrunr.jobs.annotations.Job;
+import org.jobrunr.server.runner.ThreadLocalJobContext;
+import org.jobrunr.utils.JobUtils;
+import org.springframework.core.annotation.AnnotatedElementUtils;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+
+public class JobInterceptor implements MethodInterceptor {
+    private final JobScheduler jobScheduler;
+
+    public JobInterceptor(JobScheduler jobScheduler) {
+        this.jobScheduler = jobScheduler;
+    }
+
+    @Nullable
+    @Override
+    public Object invoke(@Nonnull MethodInvocation invocation) throws Throwable {
+        Method method = invocation.getMethod();
+        Object[] args = invocation.getArguments();
+
+        // If arguments are present, throw an exception (JobRunr limitation)
+        if (args.length > 0) {
+            throw new IllegalStateException("JobRunr does not support method arguments");
+        }
+
+        // If running within a job context or method is not annotated with @Job, proceed as usual
+        if (ThreadLocalJobContext.getJobContext() != null || !AnnotatedElementUtils.hasAnnotation(method, Job.class)) {
+            return invocation.proceed();
+        }
+
+        // Extract job details
+        String declaringClassName = method.getDeclaringClass().getName();
+        String methodName = method.getName();
+        JobDetails jobDetails = new JobDetails(declaringClassName, null, methodName, new ArrayList<>());
```

Comment:
To add the parameters, you can just use `Stream.of(args).map(JobParameter::new).collect(Collectors.toList))`

Reply:
I understand this makes sense to me, I wasn't aware of this use case.
@rdehuyss - Just to clarify, do we need to throw the exception only if the method is annotated with @Recurring?

Reply:
Yes, indeed. 

But, the @Recurring is handled by the `RecurringJobPostProcessor` and out of scope for the `@AsyncJobPostProcessor`. The exception is already being thrown in the `RecurringJobPostProcessor`.

Does that make sense?

Reply:
Yes, this makes sense. Thanks for the input

Reply:
@rdehuyss - Made the changes to support parameters as well and updated the unit test for the same.